packages/server/modules/workspaces/tests/integration/invites.graph.spec.ts

```diff
@@ -598,6 +602,153 @@ describe('Workspaces Invites GQL', () => {
+        ownerId: '',
+        isPublic: false
+      }
+
+      const workspaceMemberWithNoProjectAccess: BasicTestUser = {
+        name: 'Workspace Member With No Project Access #1',
+        email: 'workspaceMemberWithNoProjectAccess1@gmail.com',
+        id: ''
+      }
+
+      before(async () => {
+        await createTestUsers([workspaceMemberWithNoProjectAccess])
+        await createTestWorkspaces([[myProjectInviteTargetWorkspace, me]])
+        await assignToWorkspaces([
+          [myProjectInviteTargetWorkspace, myWorkspaceFriend, Roles.Workspace.Member],
+          [
+            myProjectInviteTargetWorkspace,
+            workspaceMemberWithNoProjectAccess,
+            Roles.Workspace.Member
+          ]
+        ])
+
+        myProjectInviteTargetWorkspaceProject.workspaceId =
+          myProjectInviteTargetWorkspace.id
+        await createTestStreams([
+          [myProjectInviteTargetWorkspaceProject, me],
+          [myProjectInviteTargetBasicProject, me]
+        ])
+
+        // Make myworkspacefriend a project owner (but not workspace admin!)
+        await addOrUpdateStreamCollaborator(
+          myProjectInviteTargetWorkspaceProject.id,
+          myWorkspaceFriend.id,
+          Roles.Stream.Owner,
+          me.id
+        )
+
+        // Remove all project access from workspaceMemberWithNoProjectAccess
+        await Promise.all([
+          leaveStream(
+            myProjectInviteTargetWorkspaceProject,
+            workspaceMemberWithNoProjectAccess
+          ),
+          leaveStream(
+            myProjectInviteTargetBasicProject,
+            workspaceMemberWithNoProjectAccess
+          )
+        ])
+      })
+
+      afterEach(async () => {
+        await truncateTables([ServerInvites.name])
+      })
+
+      it("can't invite to workspace project through base project invite resolver", async () => {
+        const res = await gqlHelpers.createDefaultProjectInvite({
+          projectId: myProjectInviteTargetWorkspaceProject.id,
+          input: {
+            userId: otherGuy.id,
+            role: Roles.Stream.Owner
+          }
+        })
+
+        expect(res).to.haveGraphQLErrors('Target project belongs to a workspace')
+        expect(res.data?.projectMutations.invites.create.id).to.not.be.ok
+      })
+
+      it('can invite to non-workspace project through workspace project invite resolver', async () => {
+        const res = await gqlHelpers.createWorkspaceProjectInvite({
+          projectId: myProjectInviteTargetBasicProject.id,
+          inputs: [
+            {
+              userId: otherGuy.id,
+              role: Roles.Stream.Owner,
+              workspaceRole: Roles.Workspace.Admin // should be ignored
+            }
+          ]
+        })
+
+        expect(res).to.not.haveGraphQLErrors()
+        expect(res.data?.projectMutations.invites.createForWorkspace.id).to.be.ok
+      })
+
+      it("can't indirectly invite to workspace if not workspace admin", async () => {
```

Comment:
This validates: "Workspace project invites must be validated, only workspace admins can send out project invites, that also have workspace invites in them"